Bottom: 1b44c01954c574cbb54f676819e5c40b5d326726
Top:    5e6ec3084047e019e56f9159f97d239f69f3e932
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 14:59:06 +0000

XXX: THIS IS NOT TO BE EDITED, JUST A CHERRY PICKED COPY FOR DEBUG... Add polygon parentage tree debug view

NB: Also contains some probably bug fixes for PBO operations,
in particular - relating to bad exit codes when a NULL poly is
copied.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 2372153..1157244 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2895,10 +2895,8 @@ poly_Boolean_free (POLYAREA * ai, POLYAREA * bi, POLYAREA ** res, int action)
   *res = NULL;
 
   /* Make copies for tracking polygon parentage (DEBUG) */
-//  if (!poly_M_Copy0 (&a_copy, a) || !poly_M_Copy0 (&b_copy, b))
-//      return err_no_memory;
-  a_copy = NULL;
-  b_copy = NULL;
+  if (!poly_M_Copy0 (&a_copy, a) || !poly_M_Copy0 (&b_copy, b))
+      return err_no_memory;
 
   /* Move the parentage information over onto the copy */
   if (a_copy != NULL)
